### ISSUE TITLE
Some tests currently crash

### ISSUE THREAD

**Issue by Frizlab** [happn-app/XibLoc#9]
The crash happens on an assertion (internal logic error).

All the tests that fail are variants of:
```swift
let title = "yolo"
let nResults = XibLocNumber(1)
let str = "|title|^\n_#n# result<:s>_^"
let info = CommonTokensGroup(simpleReplacement1: title, simpleReplacement2: nil, number: nResults)
	.str2AttrStrXibLocInfo
	.addingSimpleSourceTypeReplacement(tokens: .init(token: "^"), replacement: { val in val })!
	.addingStringAttributesChange(
		tokens: .init(token: "_"),
		change: .changeFont(newFont: .preferredFont(forTextStyle: .caption1), preserveSizes: false, preserveBold: false, preserveItalic: false),
		allowReplace: true
	)!
var result = AttributedString("yolo\n1 result", attributes: Conf.defaultStr2AttrStrAttributes)
result.setFont(.preferredFont(forTextStyle: .caption1), range: result.index(result.startIndex, offsetByCharacters: 5)..<result.endIndex)
XCTAssertEqual(
	str.applying(xibLocInfo: info),
	result
)
```

**Comment by Frizlab** [comment]
The fix will probably need a fairly big rework on index management.
In particular, an index used in one String **cannot** be reused in another String derived from the original String.

**Comment by Frizlab** [comment]
AFAICT the crash is fixed thanks to 4b55639.
